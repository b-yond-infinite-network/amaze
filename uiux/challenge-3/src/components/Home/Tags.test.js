import Tags from './Tags';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router'
import configureMockStore from 'redux-mock-store'
import middlewares from './../../store'

const initialState = {
    home: {
        tags: ['tag1', 'tag2']
    },
    common: {
        appName: 'test appName',
        token: '1234'
    }
};

describe('Tags Component', () => {

    let mockStore;

    let tree;
    let wrapper;
    let render;

    let mockOnClickTag;

    describe('has tags', () => {

        beforeAll(() => {

            mockStore = configureMockStore(middlewares)(initialState);
            mockOnClickTag = jest.fn();

            const props = {
                tags: ['tag1', 'tag2'],
                onClickTag: mockOnClickTag
            };

            tree = <Provider store={mockStore}>
                <MemoryRouter initialEntries={["/"]}>
                    <Tags {...props} />
                </MemoryRouter>
            </Provider>;
            wrapper = mount(tree);
            console.log(wrapper.debug());
            render = renderer.create(tree).toJSON();

        });

        it('Should match the snapshot', () => {
            expect(render).toMatchSnapshot();
        });

        it('Should render tag list', () => {
            const component = wrapper.find('#tag-list');
            expect(component.length).toBe(1);
        });

    });

    describe('does not have tags', () => {

        beforeAll(() => {

            mockStore = configureMockStore(middlewares)(initialState);
            mockOnClickTag = jest.fn();

            const props = {
                tags: null,
                onClickTag: mockOnClickTag
            };

            tree = <Provider store={mockStore}>
                <MemoryRouter initialEntries={["/"]}>
                    <Tags {...props} />
                </MemoryRouter>
            </Provider>;
            wrapper = mount(tree);
            console.log(wrapper.debug());
            render = renderer.create(tree).toJSON();

        });

        it('Should match the snapshot', () => {
            expect(render).toMatchSnapshot();
        });

        it('Should NOT render tag list', () => {
            const component = wrapper.find('#tag-list');
            expect(component.length).toBe(0);
        });

    });

});