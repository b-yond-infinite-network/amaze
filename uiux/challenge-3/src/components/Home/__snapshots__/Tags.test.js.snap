// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags Component does not have tags Should match the snapshot 1`] = `
<div>
  Loading Tags...
</div>
`;

exports[`Tags Component has tags Should match the snapshot 1`] = `
<div
  className="tag-list"
  id="tag-list"
>
  <a
    className="tag-default tag-pill"
    href=""
    onClick={[Function]}
  >
    tag1
  </a>
  <a
    className="tag-default tag-pill"
    href=""
    onClick={[Function]}
  >
    tag2
  </a>
</div>
`;
